test/api-documentation.test.js

```diff
@@ -571,4 +589,237 @@ describe('<api-documentation>', function() {
       assert.notOk(node, 'method is not endered');
     });
   });
+
+  [
+    ['Compact model', true],
+    ['Regular model', false]
+  ].forEach(([name, compact]) => {
+    describe(name, () => {
+      describe('Server selection', () => {
+        let element;
+        let amf;
+
+        describe('in narrow mode', () => {
+          beforeEach(async () => {
+            amf = await AmfLoader.load(null, compact);
+            element = await narrowFixture(
+              amf,
+              'method',
+              html`
+                <anypoint-item slot="custom-base-uri" value="http://customServer.com">
+                  Server 1 - http://customServer.com
+                </anypoint-item>
+                <anypoint-item slot="custom-base-uri" value="http://customServer.com/{version}">
+                  Server 2 - http://customServer.com/{version}
+                </anypoint-item>
+              `
+            );
+
+            await nextFrame();
+          });
+
+          it('should load servers', () => {
+            assert.lengthOf(element.servers, 1);
+          });
+
+          it('should set serversCount', () => {
+            assert.equal(element.serversCount, 4);
+          });
+
+          it('should update selectedServerValue and selectedServerType using the first available server', () => {
+            assert.equal(element.selectedServerValue, 'http://{instance}.domain.com/');
+            assert.equal(element.selectedServerType, 'server');
+          });
+
+          it('should not change the baseUri property', () => {
+            assert.isUndefined(element.baseUri);
+          });
+
+          it('should render api-server-selector', () => {
+            assert.exists(element.shadowRoot.querySelector('api-server-selector'));
+          });
+
+          it('should not hide api-server-selector', () => {
+            assert.isFalse(element.shadowRoot.querySelector('api-server-selector').hidden);
+          });
+
+          describe('when navigating to something other than a method or an endpoint', () => {
+            beforeEach(async () => {
+              element.selectedType = 'summary';
+              element.selected = 'summary';
+
+              await aTimeout();
+            });
+
+            it('should hide api-server-selector', () => {
+              assert.isTrue(element.shadowRoot.querySelector('api-server-selector').hidden);
+            });
+          });
+
+          describe('when serverCount changes to less than 2', () => {
+            let serverSelector;
+
+            beforeEach(async () => {
+              serverSelector = element.shadowRoot.querySelector('api-server-selector');
```

Comment:
I think this is a good candidate for integration tests. Remove the servers from the light DOM and observe if value changed. The current way you are skipping intergation part.

Reply:
good idea, will do